pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -2322,6 +2322,23 @@ public boolean wasNull() throws SQLException {
     // varchar in binary is same as text, other binary fields are converted to their text format
     if (isBinary(columnIndex) && getSQLType(columnIndex) != Types.VARCHAR) {
       Field field = fields[columnIndex - 1];
+      TimestampUtils ts = getTimestampUtils();
+      // internalGetObject is used in getObject(int), so we can't easily alter the returned type
+      // Currently, internalGetObject delegates to getTime(), getTimestamp(), so it has issues
+      // with timezone conversions.
+      // However, as we know the explicit oids, we can do a better job here
+      switch (field.getOID()) {
+        case Oid.TIME:
+          return ts.toString(ts.toLocalTimeBin(value));
+        case Oid.TIMETZ:
+          return ts.toString(ts.toOffsetTimeBin(value));
+        case Oid.DATE:
+          return ts.toString(ts.toLocalDateBin(value));
+        case Oid.TIMESTAMP:
+          return ts.toString(ts.toLocalDateTimeBin(value));
+        case Oid.TIMESTAMPTZ:
+          return ts.toString(ts.toOffsetDateTimeBin(value));
```

Comment:
This yields `+00:00` `OffsetDateTime` since PostgreSQL does not store and return the offset.

It causes the failure like

```
    org.junit.ComparisonFailure: tstz -> getString, binary expected:<2005-01-01 1[5:00:00+03]> but was:<2005-01-01 1[2:00:00+00]>
```

Any thoughts on whether we should fix it?

Reply:
The only solution to this is to not allow sending timestamptz in binary.  Or we apply the client timezone to the received value.

Reply:
I went with "apply client timezone to the received value" (see `withOffsetSameInstant`). It keeps compatibility regarding `.getString()` which might be a nice thing to have.